src/pages/profile/index.js

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useRef } from "react";
+import { useHistory } from "react-router";
+import { SubmitButton } from "../../components/elements";
+import { AUTH } from "../../constants";
+import { getUserData, updateUserData } from "../../services/userService";
+import "./profile.css";
+
+export function Profile() {
+  const history = useHistory();
+  const nameRef = useRef();
+  const emailRef = useRef();
+  const formRef = useRef();
+  useEffect(() => {
+    getUserData()
+      .then((userData) => {
+        nameRef.current.value = userData.name;
+        emailRef.current.value = userData.email;
```

Comment:
Here you can use controlled inputs instead of using ref to update the field values. Till now we had been doing all validations on the back-end which is a must because front-end validations can be bypassed. Here let's try to validate inputs on the front-end to save one network request and our server processing.